Documentation/ApiOverview/ContentSecurityPolicy/Index.rst

```diff
@@ -464,10 +695,63 @@ ViewHelpers with the `useNonce` attribute:
         .some-style { color: red; }
     </f:asset.css>
 
+.. _content-security-policy-nonce-caching:
+
+Notes about nonces and caching
+------------------------------
+
+Nonces are implemented via a PSR middleware and thus applied dynamically. This
+also means, they are somewhat "bad" for caching (especially for reverse proxies),
+since they create unique output for a specific visitor.
+
+Since the goal of nonces are to allow "exemptions" for otherwise forbidden content,
+this closely relates to validity or integrity of this forbidden content. Instead
+of emitting unique nonces, another possibility is to utilize hashing functionality
+to content regarded as "safe".
+
+This can be done with sha256/sha384/sha512 hashing of referenced script, and including
+them as a valid directive, like this:
+
+..  code-block
+    :caption: Example YAML configuration directive, for example in config/sitepackage/csp.yaml
+
+    #...
+    - mode: "extend"
+      directive: "script-src"
+      sources:
+        - "sha256-6c7d3c1bf856597a2c8ae2ca7498cb4454a32286670b20cf36202fa578b491a9"
+
+The "sha256-..." block would be the SHA256 hash created from a file like 'script.js'.
+
+For example, a file like this:
+
+..  code-block:: javascript
+    :caption: script.js (some javascript file that is included in your website)
+
+    console.log('Hello.');
+
+would correspond to a SHA256 hash of `6c7d3c1bf856597a2c8ae2ca7498cb4454a32286670b20cf36202fa578b491a9`.
+
+..  note::
+    These hashes can be created by shell scripts like `sha256` and several libraries,
+    also in nodeJS bundling tools.
+
+The browser would evaluate a reference JavaScript file and calculate it's SHA256
+hash and compare it to the list of allowed hashes.
+
+The downside of this is: Everytime an embedded file changes (like via build processes),
+the CSP SHA hash would need to be adopted. This could be automated by a PHP definition
+of CSP rules and hashing files automatically, which would be a performance-intense
+process and call for its own caching.
+
+There is no automatism for this kind of hashing in TYPO3 (yet), so it has to be done manually
+as outlined above.
```

Comment:
Maybe link to https://forge.typo3.org/issues/100887 and https://review.typo3.org/c/Packages/TYPO3.CMS/+/80554 for the time being.

Reply:
good idea, even though we might then need to update the docs if that issue gets rejected or never implemented ;)

Reply:
Then someone will complain about it in the future ;-)